tests/workflow/test_resolve_execution_config.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for the `_resolve_execution_config` helper function in PennyLane."""
+# pylint: disable=redefined-outer-name
+import pytest
+
+import pennylane as qml
+from pennylane.devices import ExecutionConfig, MCMConfig
+from pennylane.transforms.core import TransformProgram
+from pennylane.workflow.resolution import _resolve_execution_config
+
+
+@pytest.fixture
+def tape_with_finite_shots():
+    """Fixture for creating a batch of quantum tapes with finite shots."""
+    return [qml.tape.QuantumScript([], [qml.expval(qml.Z(0))], shots=100)]
```

Comment:
If it really worth having a fixture for one line of code? 

It makes the test much more difficult to debug since I have to chase all the fixtures around to be able to replicate the failure else where.

Reply:
Fair enough! Addressed in the latest commit :)